src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof BuildData)) {
+            return false;
+        } else {
+            BuildData otherBuildData = (BuildData) o;
+
+            if (otherBuildData.remoteUrls.equals(this.remoteUrls)
```

Comment:
The tests I wrote for the equals method threw null pointer exceptions on these unguarded references to fields within the BuildData object.  I don't see any annotation which hints that they won't be null, so I assume they can be null, and this could throw a null pointer exception.